core/src/main/resources/jenkins/management/AdministrativeMonitorsDecorator/resources.js

```diff
@@ -1,78 +1,94 @@
-function amCloser(e) {
-    var list = document.getElementById('visible-am-list');
-    var el = e.target;
-    while (el) {
-        if (el === list) {
-            return; // clicked in the list
+(function () {
+    function initializeAmMonitor(amMonitorRoot, options) {
+        var button = amMonitorRoot.querySelector('.am-button');
+        var amList = amMonitorRoot.querySelector('.am-list');
+        if (button === null || amList === null) return null;
+
+        var url = button.getAttribute('data-href');
+
+        function onClose(e) {
+            var list = amList;
+            var el = e.target;
+            while (el) {
+                if (el === list) {
+                    return; // clicked in the list
+                }
+                el = el.parentElement;
+            }
+            close();
         }
-        el = el.parentElement;
-    }
-    hideVisibleAmList();
-}
-function secAmCloser(e) {
-    var list = document.getElementById('visible-sec-am-list');
-    var el = e.target;
-    while (el) {
-        if (el === list) {
-            return; // clicked in the list
+        function onEscClose(e) {
+            if (e.keyCode === 27) { // If the key press is ESCAPE...
```

Comment:
I meant something like this, what do you think?
```suggestion
            var escapeKeyCode = 27;
            if (e.keyCode === escapeKeyCode) {
```